robot/Cumulus/resources/DataImportPageObject.py

```diff
@@ -9,22 +10,68 @@ class DataImportPage(ListingPage):
     def npsp(self):
         return self.builtin.get_library_instance('NPSP')
     
+    @property
+    def pageobjects(self):
+        return self.builtin.get_library_instance("cumulusci.robotframework.PageObjects")
     
-    
-        
+    def _is_current_page(self):
+        """
+        Waits for the current page to be a Data Import list view
+        """
+        self.npsp.wait_until_url_contains("/list")
+            
     def click(self,btn_name):
         """Clicks button with specified name on Data Import Page"""
         self.npsp.click_special_object_button(btn_name)
         
-    def click_begin_data_import_process(self):
-        """On the DI page, clicks the Begin Data Import Process button and waits for batch processing to complete"""
+    def begin_data_import_process_and_verify_status(self,batch,status):
+        """On the DI page, clicks the Begin Data Import Process button and waits for specified status to display """
         self.npsp.wait_for_locator("frame","NPSP Data Import")
         self.npsp.select_frame_and_click_element("NPSP Data Import","button","Begin Data Import Process")
-        self.npsp.wait_for_batch_to_complete("data_imports.status","Completed")
+        self.npsp.wait_for_batch_to_process(batch,status)
         
     def click_close_button(self):
         """Click on close button on DI processing page and waits for DI object homepage to load"""
         self.npsp.click_button_with_value("Close")
         self.npsp.wait_until_url_contains("DataImport__c")    
         
-        
\ No newline at end of file
+    def select_import_record(self,record_name):
+        self.selenium.page_should_contain_link(record_name)
```

Comment:
Would you want to add documentation to this method as well?

Reply:
deleting this as I dont seem to use it and forgot to remove